### PR TITLE
fix: 修复 mergedCell 缺失theme 导致的报错

### LINKED ISSUE #1868: 🐛version 1.33.0 版本详情表 鼠标滚动 报错，望尽快修复
🏷 Version 1.33.0

image

**1.33.0版本有问题，详情表报错，望尽快修复。。**

1.31.0 版本没这个问题

| Package      | Version |
| -------------- | --------- |
| @antv/s2       |          |
| @antv/s2-react |          |
| @antv/s2-vue   |          |

### Sheet Type

- [ ] PivotSheet
- [x] TableSheet
- [ ] GridAnalysisSheet
- [ ] StrategySheet

### PR DESCRIPTION
**Pull request by stone-lyl**
### 👀 PR includes

✨ Feature

- [ ] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [X] Solve the issue and close #1868 

🔧 Chore

- [X] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| ❌      | ✅     |
| image |  image |
| image | image |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-core/src/theme/index.ts [hunk 1]
```diff
@@ -272,147 +413,9 @@ export const getTheme = (
...
-        size: 10,
-        margin: {
-          right: 4,
-          left: 4,
-        },
-      },
-    },
+    dataCell,
+    // ------------- MergedCell -------------------
+    mergedCell: dataCell,
```
Comment (lcx-seima) [inline packages/s2-core/src/theme/index.ts:418]: 建议不要用同一份 obj 引用 🤔

Comment (stone-lyl) [reply packages/s2-core/src/theme/index.ts:418]: 我以为这个 主题仅仅当做读取的常量使用呢，我这边 deepClone 一份吧

**Review by lijinke666 — APPROVED** [review]

**Pull request merged**